### ISSUE TITLE
Breaking changes on Neovim master branch

### ISSUE THREAD

**Issue by jose-elias-alvarez** [jose-elias-alvarez/null-ls.nvim#156]
neovim/neovim#15504 was just merged, and it contains a breaking change to LSP handler signatures. 230228b148d4f3e0c305dc3dfb08a8f993fdd839 updated null-ls to handle the change, but users who are on the Neovim master branch need to update and rebuild to avoid errors.

**Comment by kdheepak** [comment]
Hi @jose-elias-alvarez, thanks for making the changes here to match the latest neovim updates.

I'm getting the following error when neovim starts up:

```
Error executing vim.schedule lua callback: .../local/nvim/share/nvim/runtime/lua/vim/uri.lua:116: attempt to index local 'uri' (a nil value)
Press ENTER or type command to continue
```

This only happens if I have `null-ls` setup.

```lua
local null_ls = require("null-ls")

null_ls.config({
  sources = {
    null_ls.builtins.formatting.stylua.with({
      args = { "--indent-width=2", "--quote-style=AutoPreferDouble", "--indent-type=Spaces", "-" },
    }),
  },
})

lspconfig["null-ls"].setup({
  on_attach = function(client)
    if client.resolved_capabilities.document_formatting then
      vim.cmd("autocmd BufWritePre <buffer> lua vim.lsp.buf.formatting()")
    end
  end,
})
```

**Comment by vuki656** [comment]
> Hi @jose-elias-alvarez, thanks for making the changes here to match the latest neovim updates.
> 
> I'm getting the following error when neovim starts up:
> 
> ```
> Error executing vim.schedule lua callback: .../local/nvim/share/nvim/runtime/lua/vim/uri.lua:116: attempt to index local 'uri' (a nil value)
> Press ENTER or type command to continue
> ```
> 
> This only happens if I have `null-ls` setup.
> 
> ```lua
> local null_ls = require("null-ls")
> 
> null_ls.config({
>   sources = {
>     null_ls.builtins.formatting.stylua.with({
>       args = { "--indent-width=2", "--quote-style=AutoPreferDouble", "--indent-type=Spaces", "-" },
>     }),
>   },
> })
> 
> lspconfig["null-ls"].setup({
>   on_attach = function(client)
>     if client.resolved_capabilities.document_formatting then
>       vim.cmd("autocmd BufWritePre <buffer> lua vim.lsp.buf.formatting()")
>     end
>   end,
> })
> ```

If you built from source, go to the neovim folder, delete the `build` folder and run `sudo make install`. 

Fixed the issue for me.

**Comment by kdheepak** [comment]
Thanks! I did a clean build from source and it is working now. My bad!

**Comment by PrimaMateria** [comment]
Hi.
I have updated plugs with `:PlugUpdate` and I also start getting this error.
I tried to remove null-ls plugin, run `:PlugClean`, put it back and run `:PlugInstall`. The error still persists.
To fix it, do I need to remove null-js from plugin list, clone it to the `~/.config/nvim/plugged` and build it manually?

**Comment by Farkal** [comment]
There is also issue with the last released version of neovim: 0.5.1
I get error from this lines each time i change something in my files now -> https://github.com/neovim/neovim/blob/v0.5.1/runtime/lua/vim/lsp/util.lua#L253

**Comment by jose-elias-alvarez** [comment]
@Farkal This issue is specifically for the master branch, so please open an issue (and make sure to follow the template and post your config, since I don't immediately see any issues with 0.5.1).
